framework/decode/file_processor.cpp

```diff
@@ -706,6 +706,7 @@ bool FileProcessor::ProcessMetaData(const format::BlockHeader& block_header, for
                 {
                     if (decoder->SupportsMetaDataId(meta_data_id))
                     {
+                        decoder->SetCurrentBlockIndex(block_index_);
```

Comment:
These calls to `decoder->SetCurrentBlockIndex(block_index_);` are not necessary since the file processor is already calling the all registered decoders' `SetCurrentBlockIndex` from `ProcessBlocks` (line 254)